Add base functionality

rogeruiz opened the pull request:
This adds the base functionality for `coming-attractions`. As of right now, it's mostly a work in progress and completely open to discussion. Comments welcome below! 🎉 

### Acceptance Criteria:

- [x] Make it Node.
    - [x] Write some integration tests.
    - [x] Write more complete integration tests.
- [ ] Talk a bit about how node is being used for the generation of the files, style maybe?
- [ ] Talk a bit about publishing releases to npm, using npm in this way.
- [x] Make it work.
- [x] Include an `.eslintrc` and `.editorconfig`, the latter with a JS focus.
- [ ] Talk a bit about integrating this into already existing `npm` automation workflows or apps at :18f: so you have a static app you can deploy at any time.

rogeruiz committed:
Update config and scripts, initially

rogeruiz committed:
Adding in the source directory, initially

rogeruiz committed:
Update some scripts and config variables

rogeruiz committed:
Leverage npm variables

rogeruiz committed:
Update the ignorance of it all

rogeruiz committed:
Add binary dependencies

rogeruiz committed:
Add generate functionality

rogeruiz commented:
:eyes: https://coming-attractions.apps.cloud.gov/anything/but/production

Currently deployed to my `gsa-sandbox`.

As I've considered this a collaboration between DevOps and Frontend, I'm pinging a person from each guild.
cc: @linuxbozo & @xtine

rogeruiz committed:
Removing binaries

rogeruiz committed:
Remove bespoke binaries

rogeruiz committed:
Recreate the libraries in Node rather than Bash

rogeruiz committed:
Refactor mustache templates to leverage data.js

rogeruiz committed:
Refactor config, scripts, and devDependencies

rogeruiz committed:
Merge pull request #2 from 18F/make-javascript-node-again

Make javascript node again

adborden commented on source/cf/manifest.yml:
```diff
@@ -0,0 +1,11 @@
+---
```
Is there a suffix we can add to this file to indicate it's a template? Not sure if mustache has a preferred template extension but something like `manifest.yml.tpl` would be more descriptive.

rogeruiz replied:
yeah, maybe `<file_name>.<extension>.mo` ?

rogeruiz replied:
✅ https://github.com/18F/coming-attractions/pull/1/commits/09358a19412a35703c62fdfa3e4b33d127db1bf5

adborden commented:
:+1:  This could probably go in a bunch of different directions, I'm curious why not use one of the generator frameworks, like [yeoman](http://yeoman.io/)?.

adborden commented:
IIRC, the front end guild standardized on the airbnb styleguide, should we just us the airbnb `.eslintrc`?

rogeruiz commented:
@adborden mostly because of [this](https://github.com/18F/coming-attractions#development)

> In order to work on this project, it's recommended that the source code be maintained in a separate repository. It does not necessarily need to be a Github-type of fork. It can be a completely separate project with a name more appropriate for your needs.

I figured I'd just use mustache to not complicate the build process as other projects may either fork this repository to build their opinionated version or keep it as simple as possible.

rogeruiz committed:
Rename mustache templates to `.mo`

rogeruiz committed:
Initial testing setup

glbrtmrgn committed:
Fixed the preinstall issue

We separated reinstall into two hooks(preinstall and postinstall)

glbrtmrgn committed:
Abstracted the render functionality

Made render into its own function and created a simple test for this.

glbrtmrgn committed:
Refactored data

We moved the message property from generate to data.

glbrtmrgn committed:
Adding test for the generate library

Integration testing is now complete

rogeruiz committed:
Add Travis CI support for integration tests

rogeruiz committed:
Skip plugin installation if TRAVIS

rogeruiz committed:
Better expression

rogeruiz committed:
Not exiting with a nonzero status

rogeruiz committed:
Add `addons` stanza because the internet showed me

rogeruiz committed:
Add some docs around Travis CI

glbrtmrgn committed:
Merge pull request #6 from 18F/add-ci

Add Travis CI support for integration tests

rogeruiz committed:
Update deployment to a script

rogeruiz committed:
Add better documentation;

I've added some deployment documentation that might need a run through
and some writing prompts for #g-frontend contributions to help teams
make Coming Attractions in a repeatable way that might be able to stay
in sync with their current app.

rogeruiz committed:
Welcome aboard, @glbrtmrgn :wave:

rogeruiz committed:
Update docs

glbrtmrgn committed:
We added linting! :tada:

glbrtmrgn committed:
Ensured that we run lint when we test

glbrtmrgn committed:
Added the editorconfig file

glbrtmrgn committed:
Merge pull request #7 from 18F/update-docs

Update docs

rogeruiz committed:
Merge pull request #5 from 18F/finalized-1.0-release

Finalized 1.0 release

rogeruiz commented:
@glbrtmrgn will be covering the following points in separate issues.
- Talk a bit about how node is being used for the generation of the files, style maybe?
- Talk a bit about publishing releases to npm, using npm in this way.
- Talk a bit about integrating this into already existing npm automation workflows or apps at :18f: so you have a static app you can deploy at any time.

glbrtmrgn reviewed (approved):
:shipit:
